src/marbl_surface_flux_mod.F90

```diff
@@ -0,0 +1,567 @@
+    !  calculate iron and dust fluxes if necessary
+    !-----------------------------------------------------------------------
+
+    surface_fluxes(:, fe_ind) = surface_fluxes(:, fe_ind) + iron_flux_in(:)
+
+    !-----------------------------------------------------------------------
+    !  Add phosphate and silicate from dust after Krishnamurthy et al. (2010)
+    !  factors convert from g/cm2/s to nmol/cm2/s
+    !  ( P frac in dust by weight) * ( P solubility) / ( P molecular weight) * (mol->nmol)
+    !  (Si frac in dust by weight) * (Si solubility) / (Si molecular weight) * (mol->nmol)
+    !-----------------------------------------------------------------------
+
+    surface_fluxes(:, po4_ind) = surface_fluxes(:, po4_ind)   + (dust_flux_in * (0.00105_r8 *  0.15_r8 / 30.974_r8 * 1.0e9_r8))
+
+    surface_fluxes(:, sio3_ind) = surface_fluxes(:, sio3_ind) + (dust_flux_in * (  0.308_r8 * 0.075_r8 / 28.085_r8 * 1.0e9_r8))
+
+    !-----------------------------------------------------------------------
+    !  calculate nox and nhy fluxes if necessary
+    !-----------------------------------------------------------------------
+
+    if (surface_forcing_ind%nox_flux_id.ne.0) then
+      surface_fluxes(:, no3_ind) = surface_fluxes(:, no3_ind) + nox_flux(:)
+    endif
+
+    if (surface_forcing_ind%nhy_flux_id.ne.0) then
+      surface_fluxes(:, nh4_ind) = surface_fluxes(:, nh4_ind) + nhy_flux(:)
+    endif
+
+    !-----------------------------------------------------------------------
+    !  Apply NO & NH fluxes to alkalinity
+    !-----------------------------------------------------------------------
+
+    surface_fluxes(:, alk_ind)         = surface_fluxes(:, alk_ind) + &
+         surface_fluxes(:, nh4_ind) - surface_fluxes(:, no3_ind)
+    surface_fluxes(:, alk_alt_co2_ind) = surface_fluxes(:, alk_alt_co2_ind) + &
+         surface_fluxes(:, nh4_ind) - surface_fluxes(:, no3_ind)
+
+    !-----------------------------------------------------------------------
+    ! Set surface forcing diagnostics
+    !-----------------------------------------------------------------------
+
+    call marbl_diagnostics_surface_flux_compute(              &
+         surface_forcing_ind      = ind,                      &
+         surface_input_forcings   = surface_input_forcings,   &
+         surface_flux_internal    = surface_flux_internal,    &
+         marbl_tracer_indices     = marbl_tracer_indices,     &
+         saved_state              = saved_state,              &
+         saved_state_ind          = saved_state_ind,          &
+         surface_flux_diags       = surface_flux_diags)
+
+    !-----------------------------------------------------------------------
+    ! Compute carbon isotopes surface fluxes
+    !-----------------------------------------------------------------------
+
+    ! pass in sections of surface_input_forcings instead of associated vars because of problems with intel/15.0.3
```

Comment:
Having the `if (ciso_on)` check in `marbl_ciso_surface_flux_mod` means I'm always trying to pass `surface_input_forcings(ind%d1{3,4}c_id)`, and those indices are 0 for non-ciso runs. So I think we should pass all of `surface_input_forcings` (as well as `ind`) and let the subroutine pull out `sst`, `d13c`, and `d14c`.